### PR TITLE
Added some stuff

### PR DESCRIPTION
**Pull request by ryanplusplus**
- Improved name of the terminal/console make target
- Added erase make target (useful if you hose up `init.lua` and get into a boot loop)
- Improved firmware flashing by always erasing first (useful if you hose up `init.lua` and get into a boot loop) and by blanking... something?
- Improved blinky example
- Added pulsing LED example
- Added HTTP server example
- Added connection example
- Added install of `screen` on Ubuntu
- Removed Windows stubs: who are we kidding, that'll never work!

### CODE REVIEW

File: pulsy/init.lua [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    for i = 0, 1023 do
+      pwm.setduty(1, i)
+      coroutine.yield()
+    end
+
+    for i = 1023, 1, -1 do
+      pwm.setduty(1, i)
+      coroutine.yield()
+    end
+  end
```
Comment (ryanplusplus) [inline pulsy/init.lua]: I just want to say that coroutines are amazing

File: http/init.lua [hunk 2]
```diff
@@ -0,0 +1,12 @@
+net.createServer(net.TCP):listen(80, function(connection)
+  connection:on('receive', function(connection, payload)
+    local response = [[
+      'HTTP/1.0 200 OK\r\nServer: NodeMCU on ESP8266\r\nContent-Type: text/html\r\n\r\n' ..
+      '<!DOCTYPE html><html><body><h1>Hello, World!</h1></body></html>'
+    ]]
+    
+    connection:send(response, function()
+      connection:close()
+    end)
```
Comment (ryanplusplus) [inline http/init.lua]: I didn't test this, but I think this fixes the issue I was having. I will explore later.

File: pulsy/init.lua (hunk 1 above)
Comment (gvela024) [inline pulsy/init.lua]: You've explained this to me before, but it's still magic.

**Comment by gvela024** [comment]
Uhhh merge?

**Comment by ryanplusplus** [comment]
Sure, merge away. You can also wait until I've done a bit more to make sure everything works. My goal was just to communicate what's in the pipeline.

File: pulsy/init.lua (hunk 1 above)
Comment (ryanplusplus) [inline pulsy/init.lua]: I'll hit you up again tomorrow when we work on this. It's very much worth knowing.

File: Makefile [hunk 3]
```diff
@@ -28,24 +25,21 @@ install:
...
-ifeq ($(HOST),windows)
-install:
-	todo
-	sudo pip install esptool
-	sudo pip install nodemcu-uploader
-endif
-
-flash_firmware:
-	esptool.py --baud 115200 --port $(SERIAL_PORT) write_flash -fm dio -fs 4MB 0x00000 $(FIRMWARE) 0x3fc000 firmware/esp_init_data_default.bin
+flash_firmware: erase
```
Comment (ryanplusplus) [inline Makefile]: The erase goes as deep as the file system which, if it contains a broken `init.lua`, can leave the device in a boot loop even after flashing the firmware if it is not erased.

File: Makefile [hunk 4]
```diff
@@ -28,24 +25,21 @@ install:
...
 
 flash_%:
-	cd $*; nodemcu-uploader --port $(SERIAL_PORT) upload *.lua --compile
+	cd $* && nodemcu-uploader --port $(SERIAL_PORT) upload *.lua --compile
 
 erase:
+	esptool.py --baud 115200 --port $(SERIAL_PORT) erase_flash
+
+format:
 	nodemcu-uploader --port $(SERIAL_PORT) file format
```
Comment (ryanplusplus) [inline Makefile]: The old erase is now format (which just nukes the FS)

File: Makefile [hunk 5]
```diff
@@ -10,12 +10,9 @@ ifeq ($(HOST),mac)
 SERIAL_PORT?=/dev/tty.SLAB_USBtoUART
 endif
 
-ifeq ($(HOST),windows)
-SERIAL_PORT?=COM1
-endif
-
 ifeq ($(HOST),linux)
 install:
+	sudo apt install screen
```
Comment (ryanplusplus) [inline Makefile]: Apparently you don't get this out of the box

File: firmware/blank.bin [hunk 6]
```diff
@@ -0,0 +1 @@
...������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������������
```
Comment (gvela024) [inline firmware/blank.bin]: grossssssssssss

File: Makefile [hunk 7]
```diff
@@ -28,24 +25,21 @@ install:
 	sudo pip install nodemcu-uploader
 endif
 
-ifeq ($(HOST),windows)
-install:
-	todo
```
Comment (VictoriaSteele) [inline Makefile]: Need to remove `todo` from mac as well.

File: Makefile [hunk 8]
```diff
@@ -28,24 +25,21 @@ install:
...
-	sudo pip install esptool
-	sudo pip install nodemcu-uploader
-endif
-
-flash_firmware:
-	esptool.py --baud 115200 --port $(SERIAL_PORT) write_flash -fm dio -fs 4MB 0x00000 $(FIRMWARE) 0x3fc000 firmware/esp_init_data_default.bin
+flash_firmware: erase
+	esptool.py --baud 115200 --port $(SERIAL_PORT) write_flash -fm dio -fs 4MB 0x00000 $(FIRMWARE) 0x3fc000 firmware/esp_init_data_default.bin 0x7e000 firmware/blank.bin
+	echo "After flashing firmware, the filesystem may need to be formatted. This can take a while. Please be patient."
```
Comment (VictoriaSteele) [inline Makefile]: This spits out twice

**Pull request merged**